pkg/build/gobuild.go

```diff
@@ -762,7 +778,7 @@ func (g *gobuild) buildAll(ctx context.Context, s string, base v1.ImageIndex) (v
 		if err != nil {
 			return nil, err
 		}
-		img, err := g.buildOne(ctx, s, base, desc.Platform)
+		img, err := g.buildOne(ctx, ref, baseRef, base, desc.Platform)
```

Comment:
I don't think this is right.

`base.Digest()` will point to the image, but `baseRef` will point to the index, so any naive code that tries to diff for updates will always think it's changed.